### PR TITLE
Updating tests to use moq.netcore

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @Eilon \ @blowdart \ @cesarbs 

bcc @GrabYourPitchforks

File: src/Microsoft.AspNet.DataProtection/Internal/IActivator.cs [hunk 1]
```diff
@@ -3,13 +3,13 @@
 
 using System;
 
-namespace Microsoft.AspNet.DataProtection
+namespace Microsoft.AspNet.DataProtection.Internal
```
Comment (pranavkm) [inline src/Microsoft.AspNet.DataProtection/Internal/IActivator.cs]: There's currently an issue with the netcore version in that it doesn't codegen strong named assemblies. Making these internals internal instead and putting them in a `.Internal` namespace.

File: test/Microsoft.AspNet.DataProtection.Test/AuthenticatedEncryption/ConfigurationModel/AuthenticatedEncryptorDescriptorTests.cs [hunk 2]
```diff
@@ -90,9 +99,9 @@ public void CreateAuthenticatedEncryptor_RoundTripsData_ManagedImplementation(En
             var masterKey = Secret.Random(512 / 8);
             var control = new ManagedAuthenticatedEncryptor(
                 keyDerivationKey: masterKey,
-                symmetricAlgorithmFactory: () => new AesCryptoServiceProvider(),
+                symmetricAlgorithmFactory: () => Aes.Create(),
```
Comment (pranavkm) [inline test/Microsoft.AspNet.DataProtection.Test/AuthenticatedEncryption/ConfigurationModel/AuthenticatedEncryptorDescriptorTests.cs]: Missing API

File: test/Microsoft.AspNet.DataProtection.Test/RegistryPolicyResolverTests.cs [hunk 3]
```diff
@@ -195,7 +195,7 @@ public void ResolvePolicy_ManagedEncryption_WithExplicitSettings()
                 ["EncryptionType"] = "managed",
                 ["EncryptionAlgorithmType"] = typeof(TripleDES).AssemblyQualifiedName,
                 ["EncryptionAlgorithmKeySize"] = 2048,
-                ["ValidationAlgorithmType"] = typeof(HMACMD5).AssemblyQualifiedName
```
Comment (pranavkm) [inline test/Microsoft.AspNet.DataProtection.Test/RegistryPolicyResolverTests.cs]: Missing API

File: test/Microsoft.AspNet.DataProtection.Test/XmlEncryption/CertificateXmlEncryptionTests.cs [hunk 4]
```diff
@@ -1,6 +1,7 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+#if !DNXCORE50
```
Comment (pranavkm) [inline test/Microsoft.AspNet.DataProtection.Test/XmlEncryption/CertificateXmlEncryptionTests.cs]: CertificateXmlEncryption is ifdefed out.

**Comment by muratg** [comment]
cc @pakrym

**Comment by pakrym** [comment]
@pranavkm where is `moq.netcore` package?

**Comment by pranavkm** [comment]
@pakrym they aren't available on a feed as yet.

**Comment by Eilon** [comment]
Provisional :shipit: while we wait to get Moq.netcore published.

**Pull request closed without merging**